web/clusters.go

```diff
@@ -11,6 +12,182 @@ import (
+		if r.NodesRunningOn == 0 && !r.Active {
+			resource := &Resource{
+				Id: r.Id,
+			}
+			stoppedResources = append(stoppedResources, resource)
+		}
+	}
+
+	return stoppedResources
+}
+
+func (node *Node) Role() string {
+	if r, ok := node.Attributes["hana_prd_roles"]; ok {
+		role := r[strings.LastIndex(r, ":")+1:]
+		return strings.Title(role)
+	}
+	return "-"
+}
+
+func (node *Node) HealthState() string {
+	if r, ok := node.Attributes["hana_prd_roles"]; ok {
+		healthState := strings.SplitN(r, ":", 2)[0]
+		return healthState
+	}
+	return "-"
+}
+
+func (node *Node) Status() string {
+	if r, ok := node.Attributes["hana_prd_roles"]; ok {
+		status := strings.SplitN(r, ":", 3)[1]
+
+		switch status {
+		case "P":
+			return "Primary"
+		case "S":
+			return "Secondary"
+		}
+	}
+	return "-"
+}
+
+func NewNodes(c *cluster.Cluster, hl hosts.HostList) Nodes {
+	var nodes Nodes
+
+	for _, n := range c.Crmmon.NodeAttributes.Nodes {
+		node := &Node{Name: n.Name, Attributes: make(map[string]string)}
+
+		for _, a := range n.Attributes {
+			node.Attributes[a.Name] = a.Value
+		}
+
+		for _, r := range c.Crmmon.Resources {
+			if r.Node.Name == n.Name {
+				resource := &Resource{
+					Id:   r.Id,
+					Type: r.Agent,
+					Role: r.Role,
+				}
+
+				for _, p := range c.Cib.Configuration.Resources.Primitives {
+					if r.Agent == "ocf::heartbeat:IPaddr2" && r.Id == p.Id {
+						node.VirtualIps = append(node.VirtualIps, p.InstanceAttributes[0].Value)
+						break
+					}
+					switch {
+					case r.Active:
+						resource.Status = "active"
+					case r.Blocked:
+						resource.Status = "blocked"
+					case r.Failed:
+						resource.Status = "failed"
+					case r.FailureIgnored:
+						resource.Status = "failure_ignored"
+					case r.Orphaned:
+						resource.Status = "orphaned"
+					}
+				}
+
+				for _, nh := range c.Crmmon.NodeHistory.Nodes {
+					if nh.Name == n.Name {
+						for _, rh := range nh.ResourceHistory {
+							if rh.Name == resource.Id {
+								resource.FailCount = rh.FailCount
+								break
+							}
+						}
+					}
+				}
+
+				node.Resources = append(node.Resources, resource)
+			}
+		}
+
+		for _, h := range hl {
+			if h.Name() == node.Name {
+				node.Ip = h.Address
+				node.Health = h.Health()
+			}
+		}
+
+		nodes = append(nodes, node)
+	}
+
+	return nodes
+}
+
+func (nodes Nodes) SecondarySyncState() string {
+	for _, n := range nodes {
+		if n.Status() == "Secondary" {
+			if s, ok := n.Attributes["hana_prd_sync_state"]; ok {
+				return s
+			}
+		}
+	}
+	return "-"
+}
+
+func (nodes Nodes) GroupBySite() map[string]Nodes {
+	nodesBySite := make(map[string]Nodes)
+
+	for _, n := range nodes {
+		if site, ok := n.Attributes["hana_prd_site"]; ok {
+			nodesBySite[site] = append(nodesBySite[site], n)
+		}
+	}
+
+	return nodesBySite
+}
+
+func (nodes Nodes) CriticalCount() int {
+	var critical int
+
+	for _, n := range nodes {
+		if n.Health == "failed" || n.Health == "maintenance" || n.Health == "" {
```

Comment:
I wouldn't count the nodes on maintenance as critical. And I don't even know if we have the `failed` state. Is this coming from `consul` ?

Reply:
This should be `critical` not `failed`, I think it's a typo from my side.

About `maintenance`, I assumed it was a critical state by interpreting the if-then-else logic in the other templates, I'll remove the condition.